ports/tests/testindex.json

```diff
@@ -0,0 +1,101 @@
+    }],
+    "long_description" : "Just a test port written to test something.",
+    "categories"       : ["categoryA"],
+    "version"          : "1.0.0",
+    "revision"         : "0"
+},{
+    "variants"         : ["universal"],
+    "portdir"          : "categoryA\/port-A2",
+    "description"      : "This is port A2 of categoryA and category B.",
+    "homepage"         : "http:\/\/portA1.website\/",
+    "epoch"            : "0",
+    "platforms"        : "darwin",
+    "name"             : "port-A2",
+    "closedmaintainer" : true,
+    "license"          : "MIT",
+    "maintainers"      : [{
+        "email"  : {
+            "domain" : "email.com",
+            "name"   : "user"
+        }
+    }],
+    "long_description" : "Just a test port written to test something.",
+    "categories"       : ["categoryA","categoryB"],
+    "version"          : "1.0.1",
+    "revision"         : "0"
+},{
+    "variants"         : ["universal"],
+    "portdir"          : "categoryA\/port-A3",
+    "description"      : "This is port A3 of categoryA",
+    "homepage"         : "http:\/\/portA1.website\/",
+    "epoch"            : "0",
+    "platforms"        : "darwin",
+    "name"             : "port-A3",
+    "closedmaintainer" : true,
+    "license"          : "MIT",
+    "maintainers"      : [{
+        "github" : "user"
+    }],
+    "long_description" : "Just a test port written to test something.",
+    "categories"       : ["categoryA"],
+    "version"          : "1.0.0",
+    "revision"         : "0"
+},{
+    "variants"         : ["universal"],
+    "portdir"          : "categoryA\/port-A4",
+    "description"      : "This is port A1 of category A",
+    "homepage"         : "http:\/\/portA1.website\/",
+    "epoch"            : "0",
+    "platforms"        : "darwin",
+    "name"             : "port-A4",
+    "closedmaintainer" : true,
+    "license"          : "MIT",
+    "maintainers"      : [{
+        "email"  : {
+            "domain" : "email.com",
+            "name"   : "user2"
+        },
+        "github" : "user"
+    }],
+    "long_description" : "Just a test port written to test something.",
+    "categories"       : ["categoryA"],
+    "version"          : "1.0.0",
+    "revision"         : "0"
+},{
+    "variants"         : ["universal"],
+    "portdir"          : "categoryA\/port-A5",
+    "description"      : "This is port A1 of category A",
+    "homepage"         : "http:\/\/portA1.website\/",
+    "epoch"            : "0",
+    "platforms"        : "darwin",
+    "name"             : "port-A5",
+    "closedmaintainer" : true,
+    "license"          : "MIT",
+    "maintainers"      : [{
+        "email"  : {
+            "domain" : "email.com",
+            "name"   : "user"
+        },
+        "github" : "user2"
+    }],
+    "long_description" : "Just a test port written to test something.",
+    "categories"       : ["categoryA"],
+    "version"          : "1.0.0",
+    "revision"         : "0"
+}]
```

Comment:
Nitpicking: line ending (one empty line at the end) missing.

That said, for the real feedback:
* You'll need multiple such files for testing, so you should better name the file in a way that with make it easy to add further tests and still identify what this file was for (for testing maintainer import).
* I would like to see the `Portfile`s here as the source, as that's much easier and less error prone to edit. Then those `Portfile`s should be converted to some index of course. I'm not sure about the best way to do it in an elegant way (`Makefile` would be an option; not super elegant, but one way of doing the job; another option would be a simple bash or python script). Of course you would store both `Portfile`s and the json file to repository, but it should be easy (and a single command) to regenerate all the test json files from `Portfile`s.

Reply:
> * I would like to see the `Portfile`s here as the source, as that's much easier and less error prone to edit. Then those `Portfile`s should be converted to some index of course. I'm not sure about the best way to do it in an elegant way (`Makefile` would be an option; not super elegant, but one way of doing the job; another option would be a simple bash or python script). Of course you would store both `Portfile`s and the json file to repository, but it should be easy (and a single command) to regenerate all the test json files from `Portfile`s.

Okay. I just realised that automating the entire process requires `portindex2json.tcl` to output a file, while it currently just prints the entire JSON. Shall I make a PR to make `portindex2json.tcl` write to a file `portindex.json` or the web app should maintain its own separate copy? I am not sure if writing to a file will make any difference, and if it does not, then it makes sense to make a PR.

Reply:
> Okay. I just realised that automating the entire process requires `portindex2json.tcl` to output a file, while it currently just prints the entire JSON.

You can pipe the result.
```bash
portindex2json.tcl {whatever parameters are needed} > portindex.json
```
This should create a file called `portindex.json`.

Sure, we could have a flag that optionally creates a file, and if that flag is missing, outputs everything to stdout.

> Shall I make a PR to make `portindex2json.tcl` write to a file `portindex.json`

You may. But the filename should not be hardcoded (it should be a parameter), and it should not be a showstopper as piping output should still work for your purpose. Or did you experience some issues with that?

> or the web app should maintain its own separate copy?

Ideally not. If you need that script, it's better to clone it / fetch it from the official repository. If you urgently need some changes which have not yet made it into the official version, you can clone from a different repository or branch as a temporary workaround, but if you urgently need something, please complain loudly :)

Reply:
> ```shell
> portindex2json.tcl {whatever parameters are needed} > portindex.json
> ```

Thank You so much. Can't believe I totally missed it. This solves everything.